code/script.js

```diff
@@ -303,41 +323,92 @@ const filterCharacters = (keep) => {
         `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'other') {
+      if (keep) {
+        alert(
+          `Yes, the person is a smoker! Keep all smokers!`
+        )
+      }  else {
+          alert(
+            `No, the person is not a smoker, remove all smokers.`
+          )
+        }
+  } else if (category === 'hair color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(
+        `Yes, the person has ${currentQuestion.value} hair! Keep all persons with ${currentQuestion.value} hair`
+      )                     
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `No, the person doesnt have ${currentQuestion.value} hair! Remove all persons with ${currentQuestion.value} hair`
+      )                     
+    }
+  } else if (category === 'eye color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${currentQuestion.value} eyes! Keep all persons with ${currentQuestion.value} eyes`
+      )                    
+    } else {
+      alert(
+        `No, the person doesnt have ${currentQuestion.value} eyes! Remove all persons with ${currentQuestion.value} eyes`
+      )                    
     }
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+if (keep) {
+  charactersInPlay = charactersInPlay.filter((person) => person[attribute] === currentQuestion.value)
+  generateBoard(charactersInPlay)
+} else {
+  charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== currentQuestion.value)
+  generateBoard(charactersInPlay)
+  }
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  // when clicking guess, the player first have to confirm that they want to make a guess.
+  const makeAGuess = confirm(`Please confirm if you want to guess on ${suspect}`)
+  
+  if (makeAGuess === true) {
+    checkMyGuess(suspect)
+  } else {
+  }
 }
 
-// If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
```

Comment:
Great that you remembered to remove comments with suggestions from the coaches 🥳